libpromises/evalfunction.c

```diff
@@ -248,10 +248,11 @@ static JsonElement* VarRefValueToJson(EvalContext *ctx, const FnCall *fp, const
             convert = JsonArrayCreate(RlistLen(value));
             for (const Rlist *rp = value; rp != NULL; rp = rp->next)
             {
-                if (rp->val.type == RVAL_TYPE_SCALAR &&
-                    strcmp(RlistScalarValue(rp), CF_NULL_VALUE) != 0)
+                char* item = RlistScalarValueSafe(rp);
```

Comment:
Internally `ilist` is a list of strings also. They are **not** well supported in tests and in the language itself and generally just turn out to be strings. Here's an example that shows it:

```
bundle agent main
{
  vars:
      "ints" ilist => { 1, 2, 3 };
      "ints_to_json" data => '[ ints ]';
      "ints_to_json_string" string => format("%S", ints_to_json);

  reports:
      "converted ints to ${ints_to_json_string}";
}
```

The output:

```console
% cf-agent -KI -f ./test_ilist_to_json.cf                                                                                                      
R: converted ints to [["1","2","3"]]
```

Feel free to turn this into an acceptance test but I'm not sure what it proves.

> Hmmm on a second look I'm removing the whole `if` clause, since I don't want any special handling for `cf_null`.

I think you do want to kill `cf_null` if it's the only thing in the list, unless you have cleaned up every place in the code where an empty list gets converted to `{ cf_null }`. I didn't do that in my patch :)